code/game/machinery/recycle_vendor.dm

```diff
@@ -684,9 +752,23 @@
 	my_account = external_accounts["AME"]
 	for(var/machinetocheck in GLOB.machines) // link the recyclers
 		if(istype(machinetocheck, /obj/machinery/amerecycler))
-			var/obj/machinery/amerecycler/toreset = machinetocheck
-			toreset.silo = src
-			toreset.update_icon()
+			var/area/area_checked = get_area(machinetocheck)
+			var/area/this_area = get_area(src)
+			if(area_checked.vessel == this_area.vessel) // they pay taxes to the same captain
+				var/obj/machinery/amerecycler/toreset = machinetocheck
+				toreset.silo = src
+				toreset.update_icon()
+
+/obj/machinery/amesilo/proc/settledebt()
+	if(my_account.money > MINIMUM_BUDGET && taxdebt >= 100) // the taxes are paid in chunks of 100,
```

Comment:
under normal operations, taxes are paid in chunks of 100, and it must be paid in some level of chunks, as taxes are calculated as decimal values and otherwise are only rounded in one case.